pulsar-broker/src/test/java/org/apache/pulsar/client/api/DeadLetterTopicTest.java

```diff
@@ -154,28 +170,42 @@ public void testDeadLetterTopic() throws Exception {
 
         Producer<byte[]> producer = pulsarClient.newProducer(Schema.BYTES)
                 .topic(topic)
+                .enableChunking(produceLargeMessages)
+                .enableBatching(!produceLargeMessages)
                 .create();
 
+        Map<Integer, String> messageContent = new HashMap<>();
+
         for (int i = 0; i < sendMessages; i++) {
-            producer.send(String.format("Hello Pulsar [%d]", i).getBytes());
+            String data;
+            if (!produceLargeMessages) {
+                data = String.format("Hello Pulsar [%d]", i);
+            } else {
+                data = createMessagePayload(1024 * 10);
+            }
+            producer.newMessage().key(String.valueOf(i)).value(data.getBytes()).send();
+            messageContent.put(i, data);
         }
 
         producer.close();
 
         int totalReceived = 0;
         do {
             Message<byte[]> message = consumer.receive();
```

Comment:
Same as the comment https://github.com/apache/pulsar/pull/21048/files#r1309965178